src/components/AddComponent/SourceSection.tsx

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import {
+  Button,
+  ButtonVariant,
+  ExpandableSection,
+  FormGroup,
+  Grid,
+  GridItem,
+  ValidatedOptions,
+} from '@patternfly/react-core';
+import { useField, useFormikContext } from 'formik';
+import { getFieldId, HelpTooltipIcon, InputField } from '../../shared';
+import { useDebounceCallback } from '../../shared/hooks/useDebounceCallback';
+import { useFormValues } from '../form-context';
+import { AddComponentValues } from './AddComponentForm';
+import { GitAuthorization } from './GitAuthorization';
+import { GitOptions } from './GitOptions';
+import { useComponentDetection } from './utils';
+import { gitUrlRegex, containerImageRegex } from './validation-utils';
+
+type SourceSectionProps = {
+  onSamplesClick: React.MouseEventHandler<HTMLButtonElement>;
+};
+
+export const SourceSection: React.FC<SourceSectionProps> = ({ onSamplesClick }) => {
+  const [authorizationExpanded, setAuthorizationExpanded] = React.useState<boolean>(true);
+  const [showAuthorization, setShowAuthorization] = React.useState<boolean>(false);
+  const [showGitOptions, setShowGitOptions] = React.useState<boolean>(false);
+  const [, { value: source, touched, error }] = useField<string>({
+    name: 'source',
+    type: 'input',
+  });
+  const [, { value: gitOptions = {} as AddComponentValues['git'] }] =
+    useField<AddComponentValues['git']>('git');
+  const { setFieldValue } = useFormikContext();
+  const [sourceUrl, setSourceUrl] = React.useState('');
+  const [validated, setValidated] = React.useState(ValidatedOptions.default);
+  const [helpText, setHelpText] = React.useState('');
+  const [helpTextInvalid, setHelpTextInvalid] = React.useState('');
+  const [formState] = useFormValues();
+  const fieldId = getFieldId('source', 'input');
+  const isValid = !(touched && error);
+  const label = 'Git repo URL or container image';
+
+  const [detectedComponents, loadError] = useComponentDetection(
+    sourceUrl,
+    formState.application,
+    formState.namespace,
+    gitOptions.isMultiComponent,
+    gitOptions.authSecret,
+  );
+
+  const handleSourceChange = React.useCallback(() => {
+    const searchTerm = source;
+    if (!gitUrlRegex.test(searchTerm) && !containerImageRegex.test(searchTerm)) {
```

Comment:
Nit: Might be better if we store the regex tests in vars, so we don't have to test again.